src/itx.rs

```diff
@@ -409,3 +452,803 @@ inv_txfm_fn64!(32, 64, 1);
+        assign_itx_bpc_fn!($c, $pfx, $w, $h, adst_identity, H_ADST, $bpc bpc, $ext);
+        assign_itx_bpc_fn!($c, $pfx, $w, $h, flipadst_identity, H_FLIPADST, $bpc bpc, $ext);
+        assign_itx_bpc_fn!($c, $pfx, $w, $h, identity_adst, V_ADST, $bpc bpc, $ext);
+        assign_itx_bpc_fn!($c, $pfx, $w, $h, identity_flipadst, V_FLIPADST, $bpc bpc, $ext);
+    }};
+}
+
+#[cfg(all(feature = "asm", any(target_arch = "arm", target_arch = "aarch64")))]
+macro_rules! assign_itx16_fn {
+    ($c:ident, $BD:ty, $w:literal, $h:literal, $ext:ident) => {{
+        assign_itx12_fn!($c, BD, $w, $h, $ext);
+        assign_itx_fn!($c, BD, $w, $h, adst_identity, H_ADST, $ext);
+        assign_itx_fn!($c, BD, $w, $h, flipadst_identity, H_FLIPADST, $ext);
+        assign_itx_fn!($c, BD, $w, $h, identity_adst, V_ADST, $ext);
+        assign_itx_fn!($c, BD, $w, $h, identity_flipadst, V_FLIPADST, $ext);
+    }};
+
+    ($c:ident, $BD:ty, $pfx:ident, $w:literal, $h:literal, $ext:ident) => {{
+        assign_itx12_fn!($c, BD, $pfx, $w, $h, $ext);
+        assign_itx_fn!($c, BD, $pfx, $w, $h, adst_identity, H_ADST, $ext);
+        assign_itx_fn!($c, BD, $pfx, $w, $h, flipadst_identity, H_FLIPADST, $ext);
+        assign_itx_fn!($c, BD, $pfx, $w, $h, identity_adst, V_ADST, $ext);
+        assign_itx_fn!($c, BD, $pfx, $w, $h, identity_flipadst, V_FLIPADST, $ext);
+    }};
+}
+
+#[cfg(all(feature = "asm", any(target_arch = "x86", target_arch = "x86_64")))]
+#[inline(always)]
+#[rustfmt::skip]
+unsafe fn itx_dsp_init_x86<BD: BitDepth>(c: *mut Rav1dInvTxfmDSPContext, bpc: c_int) {
+
+    let flags = rav1d_get_cpu_flags();
+
+    if !flags.contains(CpuFlags::SSE2) {
+        return;
+    }
+
+    assign_itx_fn!(c, BD, 4, 4, wht_wht, WHT_WHT, sse2);
+
+    if !flags.contains(CpuFlags::SSSE3) {
+        return;
+    }
+
+    if BD::BITDEPTH == 8 {
+        assign_itx16_bpc_fn!(c,     4,  4, 8 bpc, ssse3);
+        assign_itx16_bpc_fn!(c, R,  4,  8, 8 bpc, ssse3);
+        assign_itx16_bpc_fn!(c, R,  4, 16, 8 bpc, ssse3);
+        assign_itx16_bpc_fn!(c, R,  8,  4, 8 bpc, ssse3);
+        assign_itx16_bpc_fn!(c,     8,  8, 8 bpc, ssse3);
+        assign_itx16_bpc_fn!(c, R,  8, 16, 8 bpc, ssse3);
+        assign_itx16_bpc_fn!(c, R, 16,  4, 8 bpc, ssse3);
+        assign_itx16_bpc_fn!(c, R, 16,  8, 8 bpc, ssse3);
+        assign_itx12_bpc_fn!(c,    16, 16, 8 bpc, ssse3);
+        assign_itx2_bpc_fn! (c, R,  8, 32, 8 bpc, ssse3);
+        assign_itx2_bpc_fn! (c, R, 16, 32, 8 bpc, ssse3);
+        assign_itx2_bpc_fn! (c, R, 32,  8, 8 bpc, ssse3);
+        assign_itx2_bpc_fn! (c, R, 32, 16, 8 bpc, ssse3);
+        assign_itx2_bpc_fn! (c,    32, 32, 8 bpc, ssse3);
+        assign_itx1_bpc_fn! (c, R, 16, 64, 8 bpc, ssse3);
+        assign_itx1_bpc_fn! (c, R, 32, 64, 8 bpc, ssse3);
+        assign_itx1_bpc_fn! (c, R, 64, 16, 8 bpc, ssse3);
+        assign_itx1_bpc_fn! (c, R, 64, 32, 8 bpc, ssse3);
+        assign_itx1_bpc_fn! (c,    64, 64, 8 bpc, ssse3);
```

Comment:
If you think it's worth it, it'd be nice to have these in the same order as C does.  It makes it easier to check that it's right, and easier to backport changes if new asm functions are added.

Same for the other assigning blocks like this I assume.

Reply:
Yes, I agree we should keep it in C order. 

The order you're seeing is from applying the macros step by step to the existing Rust. Somehow that diverged from the order of the C code.

Reply:
Ahh, that's weird that c2rust did that.  Oh well.